packages/amplify-gen2-codegen/src/backend/synthesizer.ts

```diff
@@ -288,6 +582,12 @@ export class BackendSynthesizer {
       );
     }
 
+    if (renderArgs.auth?.userPoolClient) {
```

Comment:
nit: Can we add a comment stating why we need to codegen this additional user pool client ? Something like:
`Since Gen2 only supports 1 user pool client by default, we need to add CDK overrides for the additional user pool client from Gen1`